metrix-integration/src/main/java/com/powsybl/metrix/integration/Metrix.java

```diff
@@ -68,32 +64,60 @@ protected void executeMetrixChunks(
             }
         }
 
-        List<CompletableFuture<?>> futures = new ArrayList<>();
-        for (int version : runParameters.getVersions()) {
+        List<CompletableFuture<?>> futures = executeVersions(analysisResult, runParameters, listener, metrixConfig, commonWorkingDir, chunkCutter);
+        if (futures.isEmpty()) {
+            return;
+        }
```

Comment:
those 3 lines are probably useless (the following for loop won't do anything if the list is empty)